docs/scoping.md

```diff
@@ -115,6 +115,66 @@ We provide some standard scope providers:
    reference to the model-class of a model-instance) is used to determine the
    concrete referenced object (e.g. the model-method, owned by a model-class).
    Example: see [tests/test_scoping/test_local_scope.py](https://github.com/igordejanovic/textX/blob/master/tests/functional/test_scoping/test_local_scope.py).
+
+   The RelativeName scope provider expects a **path through your model as string**
+   to resolve a given reference text (ref_name, in the example below,
+   "ref1=[LiteralKind]" or "ref1=[LiteralKind|ID]"): the path "a.b.c" 
+   (in the example below "ref0.vars"") denotes 
+   the path, starting form your current rule_object and yields
+   rule_object.a.b.c (this means, rule_object must have an attribute a, which 
+   in turn has an attribute b, and so on...). 
+   The **last element of your path** must either be an object containing 
+   an attribute with a name corresponding to your reference text (ref_name), 
+   or - more usual - a **list of named objects** containing the named object to 
+   be resolved.
+
+        grammar = '''
+            Model:        kinds += GroupKind values += Ref;
+            LiteralKind : name=ID;
+            GroupKind:    kindName=ID name=ID "{"
+                            vars *= LiteralKind
+                          "}";
+            Ref: ref0=[GroupKind] '.' ref1=[LiteralKind];
```

Comment:
This looks problematic to me as, if I understand correctly, you will end up with a list of `Ref` object in `Model.values` and you will have to do `ref.ref1` to get to the `LiteralKind` instance you actually wanted in the `values` list. Did I understand this correctly?

Reply:
This example demonstrates hat to simulate FQN lookup with step-by-step local scope provders:
 * **FQN lookup uses the natural structure of the model** (named things containing named things which maps to FQN-like mapping: name1.name2.name3...)
 * In my eyes: **local scope dependent lookup seeks to resolve one name** (not necessarily a FQN, typically only an ID) relative to the local context. 
  * In case of the Ref.ref1 the local context if Ref: it uses Ref.ref0 to find its targets - or in the case of Ref.ref0 it uses the parent of Ref to resolve the reference.
  * In https://github.com/igordejanovic/textX/blob/master/tests/functional/test_scoping/test_reference_to_nontextx_attribute.py you have an example with a custom provider (not RelativeName) - it is a 10-liner, including json-IO...: Here we map an ID to an entry in a JSON file...
 * The **RelativeName scope provider is just a shortcut** to help to implement local scope, like 
   * give me the method of the instance defined by the class, 
   * or (like in the new example) the configurations of the current scenario... 


--> **See new scenario example**
Maybe we find a better name for RelativeName?

Reply:
You understood correctly: You get Refs, not ref1s...
I usually use a custom class to simplify access to to ref1 (if it is a more complicated case).

Important: the example test_issue107_example_with_relative_name is not a good one (because you should better use FQN). Maybe we should remove it...

A better example is the (new) testcase/testaspect/scenario/configuration example...

Reply:
I added a comment, that you should not simulate FQN with RelativeName in the original example.

Reply:
Ah, gotcha. So, using relative name scope provider is misleading in this case. I think that we should provide tests/docs with best/recommended practice only as otherwise we could confuse users.

I agree that `test_issue107_example_with_relative_name` and `test_issue107_example_with_relative_name_deep_tree` should be converted to either use a FQN in a proper way or removed.
I also suggest the same for the docs example.

Reply:
BTW, I'm thinking in a direction of blending `FQN` and `RelativeName`. I couldn't resist the feeling that `FQN` could be just a special case of `RelativeName`. That would directly support #111

Reply:
I add the renaming of "RelativeName" to #111